### ISSUE TITLE
diffsyn代码有bug？为什么Qwen-Image-Edit-2509用DiffSynth-Studio推理比diffusers效果差很多

### ISSUE THREAD

**Issue by huxian0402** [modelscope/DiffSynth-Studio#1000]
@pengqun @Lupino @calmhawk @co63oc @wenmengzhou 
用DiffSynth-Studio最新的推理代码，相同的配置和测试集，跑出来比diffusers推理效果差很多，这个是什么原因呢？

测试图片：

![Image](https://github.com/user-attachments/assets/74ea677d-e87e-4573-ab96-94a535a7906f)
![Image](https://github.com/user-attachments/assets/50faaf82-da03-4671-bb77-f49736c0e083)

DiffSynth-Studio推理结果：

![Image](https://github.com/user-attachments/assets/6daf10ed-6793-40b0-9854-84cb0627f2b3)
![Image](https://github.com/user-attachments/assets/1668283f-2c2e-4698-a3b4-db270ada5c6f)

diffusers推理结果：

![Image](https://github.com/user-attachments/assets/5a1bc8b7-8475-4057-910b-eb31001635b1)
![Image](https://github.com/user-attachments/assets/705ea3b1-1a59-4e8f-9dd0-9d2a1ec8515d)

下面是我的测试代码：
def test_qwen_edit_diffusers():
    import os
    import torch
    from PIL import Image
    from diffusers import QwenImageEditPlusPipeline, QwenImageEditPipeline

    model_path = "Qwen-Image-Edit-2509"
    pipeline = QwenImageEditPlusPipeline.from_pretrained(model_path, torch_dtype=torch.bfloat16)
    print("pipeline loaded from", model_path)
    pipeline.to('cuda')
    pipeline.set_progress_bar_config(disable=None)

    image = Image.open("08_0.jpg").convert("RGB")
    inputs = {
        "image": image,
        "prompt": "Remove the black area.",
        "width": 1024,
        "height": 1024,
        "generator": torch.manual_seed(42),
        "true_cfg_scale": 4.0,
        "negative_prompt": "",
        "num_inference_steps": 40,
        "guidance_scale": 1.0,
        "num_images_per_prompt": 1,
    }
    with torch.inference_mode():
        output = pipeline(**inputs)
        output_image = output.images[0]

    output_image.save("qwen_edit_diffusers.jpg")
    print(f"Image saved at qwen_edit_diffusers.jpg")

def test_qwen_edit_diffsyn():
    import os, sys
    sys.path.append(os.getcwd())
    from diffsynth.pipelines.qwen_image import QwenImagePipeline, ModelConfig
    from PIL import Image
    import torch

    base_model_path = "Qwen-Image-Edit-2509"
    pipe = QwenImagePipeline.from_pretrained(
        torch_dtype=torch.bfloat16,
        device="cuda",
        model_configs=[
            ModelConfig(model_id=f"{base_model_path}", origin_file_pattern="transformer/diffusion_pytorch_model*.safetensors", skip_download=True),
            ModelConfig(model_id=f"{base_model_path}", origin_file_pattern="text_encoder/model*.safetensors", skip_download=True),
            ModelConfig(model_id=f"{base_model_path}", origin_file_pattern="vae/diffusion_pytorch_model.safetensors", skip_download=True),
        ],
        tokenizer_config=None,
        processor_config=ModelConfig(model_id=f"{base_model_path}", origin_file_pattern="processor/", skip_download=True),
    )

    image = Image.open("08_0.jpg").convert("RGB")
    output_image = pipe(
            edit_image=image,
            prompt="Remove the black area.",
            height=1024,
            width=1024,
            seed=42,
            cfg_scale=4.0,
            num_inference_steps=40, 
            negative_prompt="",
        )
    output_image.save("qwen_edit_diffsyn.jpg")
    print(f"Image saved at qwen_edit_diffsyn.jpg")

**Comment by Artiprocher** [comment]
@huxian0402 2509 是个多图编辑模型，因此即使处理单图输入，也应使用列表

```python
output_image = pipe(
        edit_image=[image], # Look at here!
        prompt="Remove the black area.",
        height=1024,
        width=1024,
        seed=42,
        cfg_scale=4.0,
        num_inference_steps=40, 
        negative_prompt="",
    )
```

这是我们的代码生成的图片：

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/8dba921f-7eb7-48a4-a2f5-07cbb5253198" />

**Comment by huxian0402** [comment]
> [@huxian0402](https://github.com/huxian0402) 2509 是个多图编辑模型，因此即使处理单图输入，也应使用列表
> 
> output_image = pipe(
>         edit_image=[image], # Look at here!
>         prompt="Remove the black area.",
>         height=1024,
>         width=1024,
>         seed=42,
>         cfg_scale=4.0,
>         num_inference_steps=40, 
>         negative_prompt="",
>     )
> 这是我们的代码生成的图片：
> 
> <img alt="Image" width="1024" height="1024" src="https://private-user-images.githubusercontent.com/35051019/503052986-8dba921f-7eb7-48a4-a2f5-07cbb5253198.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEwNDEwMDMsIm5iZiI6MTc2MTA0MDcwMywicGF0aCI6Ii8zNTA1MTAxOS81MDMwNTI5ODYtOGRiYTkyMWYtN2ViNy00OGE0LWEyZjUtMDdjYmI1MjUzMTk4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDIxVDA5NTgyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhNzNjODNhM2U0NWQwMmM1YmJjOWI4ZTk5NGNkNjJlYmViNGJlZGU5N2NlN2E1NDFiNGEwYjcwMjIzYTFjOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RqCSZGZiWSG12ahSzHIzBMIoBHAIYH3OtOt7HOUVj4E">

验证了下，是这个问题，必须传入list，才会走encode_prompt_edit_multi -> resize到384。修改后，结果和diffusers基本对齐了，谢谢大佬
